src/main/clm_initializeMod.F90

```diff
@@ -591,7 +613,7 @@ subroutine initialize2(ni,nj)
     ! Read monthly vegetation
     ! Even if CN is on, and dry-deposition is active, read CLMSP annual vegetation
     ! to get estimates of monthly LAI
-    if ( n_drydep > 0 .and. drydep_method == DD_XLND )then
+    if ( n_drydep > 0 .and. drydep_method == DD_XLND .and. .not.use_fates )then
```

Comment:
call interpMonthlyVeg(bounds_proc, canopystate_inst) really shouldn't be called when fates is on and not in SP mode.  Ultimately, I'm trying to prevent it from calling ReadMonthlyVegetation().   It will allocate arrays for lai data from 0 to maxveg. But if fates in non-sp mode is using fewer patches than there are lsmpfts, then it will overflow.

https://github.com/rgknox/ctsm/blob/fates-params-api23/src/biogeochem/SatellitePhenologyMod.F90#L398-L401

Reply:
Ahh, I see. We do need it because of the hack of how DryDep is implemented. Although I think maybe what we need to do for the immediate future is to only allow DryDep and MEGAN with FATES-SP mode. It does seem that we should be able to change it so that it only allocates the dimension size needed for the FATES case.

Reply:
This was done in my PR.